### PR TITLE
Basic Chinese support

### PR DESCRIPTION
**Pull request by gcampax**
Enough to flip it on, and start throwing sentences at it to see how it reacts.

### CODE REVIEW

File: data/test-tokenizer-zh.yml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  input: 今天下午我很忙
+  rawTokens: 今天 下午 我 很忙
+  tokens: 今天 下午 我 很忙
+  entities: {}
+
+# FIXME: CoreNLP for Chinese does not normalize times
+# see the TODO at https://github.com/stanfordnlp/CoreNLP/blob/9bcebf80930403476543f32dc20d3b875b58fd16/src/edu/stanford/nlp/ie/ChineseQuantifiableEntityNormalizer.java#L239
+#- lang: zh
+#  input: 今天四点下午我很忙
+#  rawTokens: 今天 四点 下午 我 很忙
```
Comment (johnnychhsu) [inline data/test-tokenizer-zh.yml]: Usually we say 今天下午四点我很忙

**Review by johnnychhsu — COMMENTED** [review]
So to fix the test cases, we either modify the almond-tokenizer or switch to stanfordnlp with our ner codes.
I have no idea which is better. What do you think? 
I would like to help in any case

File: data/test-tokenizer-zh.yml (hunk 1 above)
Comment (gcampax) [reply data/test-tokenizer-zh.yml]: Feel free to fix that, or add that as a separate test case.

**Comment by gcampax** [comment]
I think long-term switching to stanfordnlp might be better? I don't know, maybe we should reach out to the stanford nlp people and see which one they will maintain more.
The python package is newer, while java corenlp is battle tested and used in production by a lot of projects, but I don't know how much it is used outside of English.
And there might be other options, like spacy or nltk.

Chances are though, to fix NER you will need some custom code because ThingTalk's requirements around NER are very specific and somewhat unusual.

**Comment by johnnychhsu** [comment]
If that is the case, I think corenlp might be more appropriate for now because it is well tested. But if they will maintain stanfordnlp more, we can consider move to it.

As for the NER for Chinese, currently most of the cases I tried that failed is duration. For example, '三分十五秒', '五小時二十三分鐘'. This special cases can be handle in genie constant part ? 
Maybe we can go with corenlp now, and deal with these cases with custom codes. Depends on their maintenance, we then decide which to use?

**Comment by gcampax** [comment]
Yeah you should be able to handle those in Genie, they are just "number unit" things. The program representation will be slightly different (and potentially harder to learn) but that should be mostly fine.

Anyway if you have other tests lets push them to the branch and then merge this PR so we can move forward...

**Comment by johnnychhsu** [comment]
Sure, I will add more tests if I find some cases that fail these days.
What do you mean by program representation ? like do it in almond-tokenizer ?

**Comment by gcampax** [comment]
What I mean is that, if almond-tokenizer understands duration, a sentence of the form:
```
every 3 hours and 15 minutes post on twitter
```
is preprocessed as
```
every DURATION_0 post on twitter
```
which we would train to map to
```
timer base = now , interval = DURATION_0 => @com.twitter.post
```

If almond-tokenizer does not understand duration, and you rely on genie's understanding of measures, the sentence would be preprocessed as
```
every NUMBER_0 hours and NUMBER_1 minutes post on twitter
```
which Genie would train to map to
```
timer base = now , interval = NUMBER_0 unit:h + NUMBER_1 unit:min => @com.twitter.post
```

The latter program is more complicated, which means that the parser has to do more work, to compensate for a weaker tokenizer.

**Comment by johnnychhsu** [comment]
I see. Thank you!
BTW I don't have other test cases now, that's temporarily all I have.
After we add more template, we could have more...

**Comment by johnnychhsu** [comment]
@gcampax 
I think we can merge this branch now.
Could add more tests when we add templates.

**Pull request merged**